src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -124,13 +130,185 @@ Status RedisStoreClient::AsyncDelete(const std::string &table_name,
   return shard_context->RunArgvAsync(args, delete_callback);
 }
 
+Status RedisStoreClient::AsyncBatchDelete(const std::string &table_name,
+                                          const std::vector<std::string> &keys,
+                                          const StatusCallback &callback) {
+  auto finished_count = std::make_shared<int>(0);
+  int size = keys.size();
+  for (auto &key : keys) {
+    auto done = [finished_count, size, callback](const Status &status) {
+      ++(*finished_count);
+      if (*finished_count == size) {
+        callback(Status::OK());
+      }
+    };
+    RAY_CHECK_OK(AsyncDelete(table_name, key, done));
+  }
+  return Status::OK();
+}
+
 Status RedisStoreClient::AsyncDeleteByIndex(const std::string &table_name,
                                             const std::string &index_key,
                                             const StatusCallback &callback) {
-  RAY_CHECK(0) << "Not implemented! Will implement this function in next PR.";
+  std::string match_pattern = index_key + table_name + "*";
+  auto scanner = std::make_shared<RedisScanner>(redis_client_, table_name, match_pattern);
+  auto on_done = [this, table_name, index_key, callback, scanner](
+                     const Status &status, const std::vector<std::string> &result) {
+    if (!result.empty()) {
+      auto pos = index_key.size() + table_name.size();
+      std::vector<std::string> keys;
+      keys.resize(result.size());
+      for (auto &item : result) {
+        keys.push_back(item.substr(pos, item.size() - pos));
```

Comment:
What is each item in `result`? is it just one key or is it still the concatenation of multiple keys?
Would be better to use the former one.

Reply:
Each item in result is table key.